StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -438,6 +441,9 @@ extension EmbeddedPaymentElement: EmbeddedFormViewControllerDelegate {
             }
         }
         embeddedFormViewController.dismiss(animated: true)
+        if case .immediateAction(let didSelectPaymentOption) = configuration.rowSelectionBehavior {
+            didSelectPaymentOption()
+        }
```

Comment:
Should this work be done in the completion of dismissing?

Also, just to be clear, didSelectPaymentOption() always gets fired while `informDelegateIfPaymentOptionUpdated` works ONLY when the payment method changes, correct?  Feels awkward to sometimes communicate paymentOption via a callback sometimes, and sometimes through a delegate, but i'm guessing it was already discussed during api review?

Reply:
re: Should this work be done in the completion of dismissing?

That's an interesting idea, I don't feel too strongly but my general sense is that as soon as the user taps "Continue" we should invoke the callback. This is also consistent with how we inform the delegate, so I'm inclined to leave as-is.

re: but i'm guessing it was already discussed during api review?

Yep, there were some iterations on this, but landed on this approach.